pkg/userdata/centos/provider.go

```diff
@@ -277,6 +290,19 @@ write_files:
   content: |
 {{ containerRuntimeHealthCheckSystemdUnit | indent 4 }}
 
+{{- if .NodeImageRegistry }}
+- path: /run/containers/registries.conf
+  permissions: "0644"
+  content: |
+    INSECURE_REGISTRY="--insecure-registry {{ .NodeImageRegistry }}"
```

Comment:
Uhm. That this configures the registry as insecure should be at the very least documented and was not all obvious to me when reading the readme.

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/559/commits/afd9104ec6e27674aaf8885e1a8e107b8c309e48